src/openqaoa-qiskit/tests/test_qiskit_backend_wrapper.py

```diff
@@ -264,5 +264,160 @@ def test_expectation(self):
+
+
+    def test_zne_get_counts(self):
+        """
+        Testing the get_counts method of the ZNE wrapper.
+        """
+        self.factory = "Linear"
+        self.scaling = "fold_gates_at_random"
+        self.scale_factors = [1, 1]
+
+        self.order = 1
+        self.steps = 1
+
+        self.qaoa_descriptor, self.variate_params = get_params()
+
+        self.qiskit_shot_backend = QAOAQiskitBackendShotBasedSimulator(
+            qaoa_descriptor=self.qaoa_descriptor,
+            n_shots=100,
+            prepend_state=None,
+            append_state=None,
+            init_hadamard=True,
+            cvar_alpha=1.0,
+            qiskit_simulation_method="automatic",
+            seed_simulator=2642,
+            noise_model=None,
+        )
+
+        try:
+            self.wrapped_obj_zne = ZNEWrapper(
+                self.qiskit_shot_backend,
+                factory=self.factory,
+                scaling=self.scaling,
+                scale_factors=self.scale_factors,
+                order=self.order,
+                steps=self.steps,
+            )
+            #self.wrapped_obj_trivial
+        except NameError:
+            print("Skipping Qiskit tests as Qiskit is not installed.")
+            return None
+        qc =  self.qiskit_shot_backend.qaoa_circuit(self.variate_params)
+
+        assert self.wrapped_obj_zne.get_counts(
+            qc, n_shots=100,
+        ) == {
+            "01": 35,
+            "10": 29,
+            "11": 36
+        }, "The get_counts function in the wrapper didn't return the expected counts."
+
+    def test_expectation_zne(self):
+        """
+        Testing the expectation method of the ZNE wrapper which overrides the backend function as defined in basebackends.
+        Also, the trivial case compared with the original backend
+        """
+        self.factory = "Linear"
+        self.scaling = "fold_gates_at_random"
+        self.scale_factors = [1, 1]
+
+        self.order = 1
+        self.steps = 1
+
+        self.qaoa_descriptor, self.variate_params = get_params()
+
+        self.qiskit_shot_backend = QAOAQiskitBackendShotBasedSimulator(
+            qaoa_descriptor=self.qaoa_descriptor,
+            n_shots=10000,
+            prepend_state=None,
+            append_state=None,
+            init_hadamard=True,
+            cvar_alpha=1.0,
+            qiskit_simulation_method="automatic",
+            seed_simulator=2642,
+            noise_model=None,
+        )
+        try:
+            self.wrapped_obj_zne = ZNEWrapper(
+                self.qiskit_shot_backend,
+                factory=self.factory,
+                scaling=self.scaling,
+                scale_factors=self.scale_factors,
+                order=self.order,
+                steps=self.steps,
+            )
+        except NameError:
+            print("Skipping Qiskit tests as Qiskit is not installed.")
+            return None
```

Comment:
If Qiskit is the only backend supported, does it make sense to ignore an import error here?